packages/opentelemetry-types/src/context/propagation/Propagation.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SpanContext } from '../../trace/span_context';
+
+/** Defines a propagation interface. */
+export interface Propagation {
+  /**
+   * Injects the given {@link SpanContext} instance to transmit over the wire.
+   *
+   * OpenTelemetry defines a common set of format values (BinaryFormat and
+   * HTTPTextFormat), and each has an expected `carrier` type.
```

Comment:
Propagation can be done on non-HTTP protocols. There should be a distinction between HTTP headers and text map, similar to OpenTracing.

Reply:
This interface is inspired from [OpenTracing](https://github.com/opentracing/opentracing-javascript/blob/9aaa299d92a737dbfeabdabff96a5831ca548f7b/src/tracer.ts#L124) codebase and OTel [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/propagators-api.md). Let me know if I missed anything here.